n2y/database.py

```diff
@@ -42,9 +43,25 @@ def children(self):
         return self._children
 
     def children_filtered(self, filter, sort=None):
-        if self._children is None:
-            self._children = self.client.get_database_pages(self.notion_id, filter, sort)
-        return self._children
+        if not filter:
```

Comment:
What was the motivation for these changes?

Reply:
If the database is pulled from the cache and a filter has already been run on it,  but you want it to run a different filter it won't. Thus, documents/510k was just giving me the same results as documents/dhf.

Reply:
Can we add an intermediate variable like `loading_from_cache = filter != None` or something similar to capture why we're branching?